platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -57,3 +57,39 @@ include::sample[dir="snippets/bestPractices/usePluginsBlock/groovy/do-this",file
 
 === Tags
 `#structuring-builds`
+
+[[avoid_duplicate_dependencies]]
+== Avoid Declaring Dependencies Multiple Times
+
+You should avoid duplicating dependencies in your build scripts, whether with different versions or in different dependency configurations.
+
+=== Explanation
+
+Duplicating dependencies in build scripts can lead to several problems:
+
+* Conflicting versions may cause unexpected behavior or runtime errors due to version mismatches
```

Comment:
Like my comment below, conflicting versions are different dependencies.  Also, the example doesn't show a _conflict_, which is something that would fail a build, but merely a successful build that is kind of "irrational" in what it's requesting.

Reply:
>  Also, the example doesn't show a conflict, which is something that would fail a build, but merely a successful build that is kind of "irrational" in what it's requesting.

So you'd prefer to show the exact case when it would fail? e.g. display some code that works on version A and doesn't on version B?

Reply:
I think the idea was to show multiple declarations for the same version, possibly one direct and one transitive - not 2 declarations for different versions that conflict.

So I was thinking of a case where the build _would not fail_, but the multiple declarations would still be unwanted.

Reply:
w.r.t transitive dependency and its direct duplicate. I think it is OK to have it.
Especially if user is explicit about it. 
e.g. I can have a ktor as a dependency. Which transitively brings some parts of kotlinx coroutines as API. And I'd like to use coroutines as well. So here I think it is logical to not rely on the fact that ktor exports it as well, but declare it explicitly.

And things got spicier when suddenly an upgrade of Ktor may upgrade my coroutines as well. But that is a topic of different BP.


So let me know if you're ok with the current example. feel free to resolve this thread if everything is ok.

Reply:
I suppose it depends on the particular case.  I would say if your code needs a dep to compile, its okay to declare it directly, even if its already a transitive dep of another one of your deps.  But if you only need it because your direct deps require it, then you shouldn't declare it directly.

Reply:
I think this example is okay now.